### PR TITLE
Use the sampling period from GPU counter spec descriptor.

### PR DESCRIPTION
**Pull request by lpy**
Previously AGI hard coded the min sampling period to 1ms and max sampling
period to 1000ms. This patch changes that to use the sampling period from the
GPU counter spec descriptor and modifies the time unit to microsecond in order
to allow higher sampling rate.

Bug: b/154149534

### CODE REVIEW

File: gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java [hunk 1]
```diff
@@ -492,8 +494,26 @@ public BasicInputArea(Composite parent, Settings settings, Theme theme,
               withIndents(new GridData(), GROUP_INDENT, 0));
           gpuCountersLabels = new Label[3];
           gpuCountersLabels[1] = createLabel(counterGroup, "Poll Rate:");
-          gpuCountersRate = createSpinner(counterGroup, sGpu.getCounterRate(), 1, 1000);
-          gpuCountersLabels[2] = createLabel(counterGroup, "ms");
+
+          long minSamplingPeriod = NANOSECONDS.toMicros(gpuCaps.getGpuCounterDescriptor().getMinSamplingPeriodNs());
+          long maxSamplingPeriod = gpuCaps.getGpuCounterDescriptor().getMaxSamplingPeriodNs();
```
Comment (hevrard) [inline gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java:499]: Shouldn't you also have NANOSECONDS.toMicros(...) around ...getMaxSamplingPeriod**Ns**() ?

**Comment by lpy** [comment]
ptal

File: gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java [hunk 2]
```diff
@@ -492,8 +494,26 @@ public BasicInputArea(Composite parent, Settings settings, Theme theme,
               withIndents(new GridData(), GROUP_INDENT, 0));
           gpuCountersLabels = new Label[3];
           gpuCountersLabels[1] = createLabel(counterGroup, "Poll Rate:");
-          gpuCountersRate = createSpinner(counterGroup, sGpu.getCounterRate(), 1, 1000);
-          gpuCountersLabels[2] = createLabel(counterGroup, "ms");
+
+          long minSamplingPeriod = NANOSECONDS.toMicros(gpuCaps.getGpuCounterDescriptor().getMinSamplingPeriodNs());
+          long maxSamplingPeriod = NANOSECONDS.toMicros(gpuCaps.getGpuCounterDescriptor().getMaxSamplingPeriodNs());
+          maxSamplingPeriod = maxSamplingPeriod > 0 ? maxSamplingPeriod : minSamplingPeriod * 1000;
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java:500]: what if the min sampling period is also 0? we should first set the min sampling period to 1ms if it's not specified.

Comment (lpy) [reply gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java:500]: You are right. I plan to add validation and then notify vendors although I think they all at least advertise min sampling period. In that case do we still need to set the min sampling period to a default value?

Comment (lpy) [reply gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java:500]: ping for feedback

Comment (silence-do-good) [reply gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java:500]: Maybe we can have a min value for now and add a todo for when the validation part is completed. When it has become a requirement that they should set min value, we can remove the hack then.

Comment (pmuetschard) [reply gapic/src/main/com/google/gapid/perfetto/views/TraceConfigDialog.java:500]: I wouldn't call it a hack. We have a default for the max, it makes sense to have a default for the min. I would prefer a default we expect to never use over bad behavior we expect to never happen.

**Comment by lpy** [comment]
Thanks for the feedback, ptal!

**Pull request merged**